### PR TITLE
manual: how to add new search engines and change the default one

### PR DESCRIPTION
**Pull request by Blade6570**
I have updated the manual according to our disscussion in https://github.com/atlas-engineer/nyxt/issues/1554

### CODE REVIEW

File: source/manual.lisp [hunk 1]
```diff
@@ -104,6 +104,30 @@ keybindings have priorities over the other modes.")
     (:h3 "Search engines")
     (:p "See the " (:code "search-engines") " buffer slot documentation.
 Bookmarks can also be used as search engines, see the corresponding section.")
+    (:p "Nyxt comes with two default broswers as duckduckgo and wiki. you can append new search engines easily using the following template.")
```
Comment (Ambrevar) [inline source/manual.lisp:107]: Typo: "browsers".

Style: "Nyxt comes with two default browsers: DuckDuckGo and Wikipedia".

File: source/manual.lisp [hunk 2]
```diff
@@ -104,6 +104,30 @@ keybindings have priorities over the other modes.")
...
+    (:pre (:code "
+\(defvar `*my-search-engines*'
+\"list of search engines.\"
+  (list
+   '(\"google\" \"https://www.google.com/search?q=~a\" \"https://www.google.com/\")\
```
Comment (Ambrevar) [inline source/manual.lisp:112]: I'd rather leave out privacy-intrusive services like Google.  One of the goals of Nyxt is to protect the user's privacy after all :)

File: source/manual.lisp [hunk 3]
```diff
@@ -104,6 +104,30 @@ keybindings have priorities over the other modes.")
...
+   '(\"python3\" \"https://docs.python.org/3/search.html?q=~a\" \"https://docs.python.org/3\")\
+   '(\"doi\" \"https://dx.doi.org/~a\" \"https://dx.doi.org/\")\))
+
+(define-configuration buffer
+  ((search-engines (append (mapcar (lambda (engine) (apply 'make-search-engine engine))
+                                   *my-search-engines*)
+                           %slot-default%))))"))
+    (:p " It is to be noted that the last member in the search-engines is the default search engine of the browser. For example, in order to make google as the default browser, the above code can slightly be modified as follows.")
```
Comment (Ambrevar) [inline source/manual.lisp:120]: Break this string to avoid too long lines.

File: source/manual.lisp [hunk 4]
```diff
@@ -104,6 +104,30 @@ keybindings have priorities over the other modes.")
...
+        (:pre (:code "
+\(defvar *my-search-engines*
+  (list
+   '(\"python3\" \"https://docs.python.org/3/search.html?q=~a\" \"https://docs.python.org/3\")
+   '(\"doi\" \"https://dx.doi.org/~a\" \"https://dx.doi.org/\")
+   '(\"google\" \"https://www.google.com/search?q=~a\" \"https://www.google.com/\")))
+
+(define-configuration buffer
+  ((search-engines (append %slot-default% (mapcar (lambda (engine) (apply 'make-search-engine engine))
+                                   *my-search-engines*)))))"))
```
Comment (Ambrevar) [inline source/manual.lisp:130]: Indentation of this last line seems wrong.

**Comment by Ambrevar** [comment]
Welcome to the work of Nyxt contributors :)

Thank you very much for this documentation improvement, the doc is what needs the most focus at the moment!

Can you address the points I've raised?  Then I'll be happy to merge!

File: source/manual.lisp (hunk 1 above)
Comment (Blade6570) [reply source/manual.lisp]: I changed it to "Nyxt comes with two default search engines: DuckDuckGo and Wikipedia". browsers seems to be wrong here.

**Comment by Blade6570** [comment]
Hi,

Thanks for the feedbacks. I have fixed the issues raised by you. Have a look and let me know if you need anything else :)

File: source/manual.lisp [hunk 5]
```diff
@@ -104,6 +104,31 @@ keybindings have priorities over the other modes.")
     (:h3 "Search engines")
     (:p "See the " (:code "search-engines") " buffer slot documentation.
 Bookmarks can also be used as search engines, see the corresponding section.")
+    (:p "Nyxt comes with two default search engines: DuckDuckGo and Wikipedia.
```
Comment (edgar-vincent) [inline source/manual.lisp:107]: Should we write "Nyxt comes with two search engines by default" instead? The current phrasing might cause confusion with the "default search engine" as the search engine which is used by default when a search is started.

Comment (Blade6570) [reply source/manual.lisp:107]: yes "Nyxt comes with two search engines by default: X and Y" sounds better. I will change it.

Comment (Blade6570) [reply source/manual.lisp]: Okay it is there now :)

**Comment by Ambrevar** [comment]
I accidently merged without squashing :(

**Comment by jmercouris** [comment]
HOW COULD YOU!? Unbelievable. /s :-D

**Comment by Ambrevar** [comment]
Yup, failed to see the 4 commits in the pull request...
Anyways, I've edited this change a bit, see c9be9d8f3d756e294f3a82bb87c55b78cbfbfed0.

**Pull request merged**